### PR TITLE
Working HDFS containers.

### PR DESCRIPTION
**Pull request by fnothaft**
These two commits add working HDFS containers (Hadoop 2.6.2), that have been tested and validated to work with Spark 1.5.2 and ADAM.

### CODE REVIEW

**Comment by jvivian** [comment]
Jenkins, can you test this please?

File: adam/runtime/wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+HADOOP_PREFIX=/opt/apache-hadoop
+HADOOP_CONF_DIR=${HADOOP_PREFIX}/etc/hadoop
+ln -s ${HADOOP_PREFIX} /usr/local/hadoop
+
+# overwrite hostname in conf
+sed "s/HOSTNAME/${YARN_MASTER:-localhost}/g" $HADOOP_PREFIX/etc/hadoop/core-site.xml.template > $HADOOP_PREFIX/etc/hadoop/core-site.xml
```
Comment (heuermh) [inline adam/runtime/wrapper.sh]: is `YARN_MASTER` set anywhere?

Comment (fnothaft) [inline adam/runtime/wrapper.sh]: This doesn't get used (since we're no longer setting up YARN), but YARN_MASTER --> localhost if unset.

I'll remove this.

Comment (heuermh) [inline adam/runtime/wrapper.sh]: Great, thanks

**Comment by heuermh** [comment]
Sorry didn't catch this before, hadn't run `make clean`.  Perhaps Jenkins would've caught it, had it run.

```
apache-hadoop-common master
$ make
cd build && docker build -t computationalgenomicslab/apache-hadoop-common:2.6.2-build .
Sending build context to Docker daemon  2.56 kB
Step 1 : FROM ubuntu
 ---> 91e54dfb1179
...
Removing intermediate container bfea0e35b786
Successfully built bcc2e845a3f1
echo "Moving apache-hadoop from container to "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime""
Moving apache-hadoop from container to /Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime
docker run -v "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime":/data computationalgenomicslab/apache-hadoop-common:2.6.2-build cp -rf /opt/apache-hadoop /data/
# tomcat files must be writable
chmod ugoa+rw -R "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime"/apache-hadoop/share/hadoop/*/tomcat/
chmod: -R: No such file or directory
make: *** [runtime/apache-hadoop] Error 1
```

**Comment by fnothaft** [comment]
@heuermh Can you `ls "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime"/apache-hadoop/share/hadoop/*/tomcat/`?

**Comment by heuermh** [comment]
``` bash
apache-hadoop-common master
$ ls -als "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime"/apache-hadoop/share/hadoop/*/tomcat/
/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime/apache-hadoop/share/hadoop/httpfs/tomcat/:
total 176
  0 drwxrwxrwx  13 heuermh  staff    442 Dec 10 17:24 .
  0 drwxr-xr-x   3 heuermh  staff    102 Dec 10 17:24 ..
112 -rw-r--r--   1 heuermh  staff  56653 Dec 10 17:24 LICENSE
  8 -rw-r--r--   1 heuermh  staff    761 Dec 10 17:24 NOTICE
 24 -rw-r--r--   1 heuermh  staff   9360 Dec 10 17:24 RELEASE-NOTES
 32 -rw-r--r--   1 heuermh  staff  16229 Dec 10 17:24 RUNNING.txt
  0 drwxr-xr-x  24 heuermh  staff    816 Dec 10 17:24 bin
  0 drwxr-xr-x  10 heuermh  staff    340 Dec 10 17:24 conf
  0 drwxr-xr-x  18 heuermh  staff    612 Dec 10 17:24 lib
  0 drwxr-xr-x   2 heuermh  staff     68 Dec 10 17:24 logs
  0 drwxr-xr-x   3 heuermh  staff    102 Dec 10 17:24 temp
  0 drwxr-xr-x   4 heuermh  staff    136 Dec 10 17:24 webapps
  0 drwxr-xr-x   2 heuermh  staff     68 Dec 10 17:24 work

/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime/apache-hadoop/share/hadoop/kms/tomcat/:
total 176
  0 drwxrwxrwx  13 heuermh  staff    442 Dec 10 17:24 .
  0 drwxr-xr-x   3 heuermh  staff    102 Dec 10 17:24 ..
112 -rw-r--r--   1 heuermh  staff  56653 Dec 10 17:24 LICENSE
  8 -rw-r--r--   1 heuermh  staff    761 Dec 10 17:24 NOTICE
 24 -rw-r--r--   1 heuermh  staff   9360 Dec 10 17:24 RELEASE-NOTES
 32 -rw-r--r--   1 heuermh  staff  16229 Dec 10 17:24 RUNNING.txt
  0 drwxr-xr-x  24 heuermh  staff    816 Dec 10 17:24 bin
  0 drwxr-xr-x  10 heuermh  staff    340 Dec 10 17:24 conf
  0 drwxr-xr-x  18 heuermh  staff    612 Dec 10 17:24 lib
  0 drwxr-xr-x   2 heuermh  staff     68 Dec 10 17:24 logs
  0 drwxr-xr-x   3 heuermh  staff    102 Dec 10 17:24 temp
  0 drwxr-xr-x   4 heuermh  staff    136 Dec 10 17:24 webapps
  0 drwxr-xr-x   2 heuermh  staff     68 Dec 10 17:24 work
```

**Comment by heuermh** [comment]
Might just be an OSX thing

``` bash
apache-hadoop-common master
$ chmod ugoa+rw -R "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime"/apache-hadoop/share/hadoop/*/tomcat/
chmod: -R: No such file or directory

apache-hadoop-common master
$ chmod -R ugoa+rw "/Users/heuermh/working/cgl-docker-lib/apache-hadoop-common/runtime"/apache-hadoop/share/hadoop/*/tomcat/
```

**Pull request merged**